app/routes.py

```diff
@@ -1,2 +1,197 @@
+    # Consider using the keyword arguments data, json, and/or headers
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    url = "https://slack.com/api/chat.postMessage"
+    data = {
+        "channel": "C021ACT6H0B",
+        "text": (f"Someone just completed the task {task.title}")
+    }
+    headers = {
+        "Authorization": f"Bearer {slack_key}"
+    }
+
+    r = requests.post(url, data=data, headers=headers)
+
+    return make_response({"task": task.to_json()})
+
+
+@task_list_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+    else:
+        task.completed_at = None
+        db.session.commit()
+
+    return make_response({"task": task.to_json()})
+
+
+# WAVE 5 & 6 GOALS
+
+@goal_list_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    else:
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit() 
+    
+    # returning single task created by calling to_json on new task
+    return make_response({"goal": new_goal.to_json()}, 201)
+    
+
+@goal_list_bp.route("", methods=["GET"])
+def get_all_goals():
+    goals = Goal.query.all()
+    goals_response = []
+    for goal in goals:
+        # building a list of jsons by calling to_json on each task
+        goals_response.append(goal.to_json())
+
+    return jsonify(goals_response, 200)
+
+
+@goal_list_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def goal(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return make_response("", 404)
+    elif request.method == "GET":
+        return make_response({"goal": goal.to_json()}, 200)
+    
+    elif request.method == "PUT":
+        goal_data = request.get_json()
+        goal.title = goal_data["title"]
+
+        db.session.commit()
+
+        return make_response({"goal": goal.to_json()}), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        
+        return make_response({"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'})
+
+
+@goal_list_bp.route("<goal_id>/tasks", methods=["POST", "GET"])
+def tasks_in_goal(goal_id):
+
+    goal = Goal.query.get(goal_id)
+
+    if goal is None:
+        return ("", 404)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        for task_id in request_body["task_ids"]:
+            task = Task.query.get(task_id)
+            task.goal_offspring = goal.goal_id
+        
+        return make_response({
+            "id": goal.goal_id,
+            "task_ids": request_body["task_ids"]}, 200)
```

Comment:
This change would have made the above-mentioned bug more obvious. This would also allow the response to include ALL tasks associated with the goal, not just the new ones.
```suggestion
        task_list = []
        for task in goal.tasks:
            task_list.append(task.task_id)
        
        return make_response({
            "id": goal.goal_id,
            "task_ids": task_list}, 200)
```

Reply:
This isn't super intuitive so I'd definitely be down to talk through this more in our next 1:1